sdk/core/azure-core/src/main/java/com/azure/core/util/SharedExecutorService.java

```diff
@@ -0,0 +1,352 @@
+        return false;
+    }
+
+    /**
+     * Checks if the executor service is terminated.
+     * <p>
+     * Will always return false as the shared executor service cannot be terminated.
+     *
+     * @return False, as the shared executor service cannot be terminated.
+     */
+    @Override
+    public boolean isTerminated() {
+        return false;
+    }
+
+    @Override
+    public boolean awaitTermination(long timeout, TimeUnit unit) {
+        throw LOGGER.logThrowableAsError(
+            new UnsupportedOperationException("This executor service is shared and cannot be terminated."));
+    }
+
+    @Override
+    public void execute(Runnable command) {
+        ensureNotShutdown().execute(command);
+    }
+
+    @Override
+    public <T> Future<T> submit(Callable<T> task) {
+        return ensureNotShutdown().submit(task);
+    }
+
+    @Override
+    public <T> Future<T> submit(Runnable task, T result) {
+        return ensureNotShutdown().submit(task, result);
+    }
+
+    @Override
+    public Future<?> submit(Runnable task) {
+        return ensureNotShutdown().submit(task);
+    }
+
+    @Override
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> tasks) throws InterruptedException {
+        return ensureNotShutdown().invokeAll(tasks);
+    }
+
+    @Override
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> tasks, long timeout, TimeUnit unit)
+        throws InterruptedException {
+        return ensureNotShutdown().invokeAll(tasks, timeout, unit);
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks) throws InterruptedException, ExecutionException {
+        return ensureNotShutdown().invokeAny(tasks);
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> tasks, long timeout, TimeUnit unit)
+        throws InterruptedException, ExecutionException, TimeoutException {
+        return ensureNotShutdown().invokeAny(tasks, timeout, unit);
+    }
+
+    private ExecutorService ensureNotShutdown() {
+        return wrappedExecutorService.updateAndGet(wrapper -> {
+            if (wrapper == null || wrapper.executorService.isShutdown() || wrapper.executorService.isTerminated()) {
+                return createSharedExecutor();
+            } else {
+                return wrapper;
+            }
+        }).executorService;
+    }
+
+    private static ExecutorWithMetadata createSharedExecutor() {
+        ThreadFactory threadFactory;
+        if (VIRTUAL_THREAD_SUPPORTED && THREAD_POOL_VIRTUAL) {
+            try {
+                LOGGER.verbose("Attempting to create a virtual thread factory.");
+                threadFactory = createVirtualThreadFactory();
+                LOGGER.verbose("Successfully created a virtual thread factory.");
+            } catch (Exception e) {
+                LOGGER.info("Failed to create a virtual thread factory, falling back to non-virtual threads.", e);
```

Comment:
Should this be `verbose` too? Logging an exception as info (most customers use this log level) can trigger alerts in customer apps.

Reply:
I'd go with INFO here as this is invalid configuration with a fallback, the others are verbose when the behavior is expected.